src/ray/gcs/accessor.h

```diff
@@ -772,6 +772,35 @@ class PlacementGroupInfoAccessor {
   PlacementGroupInfoAccessor() = default;
 };
 
+class InternalKVAccessor {
+ public:
+  virtual ~InternalKVAccessor() = default;
+
+  virtual Status AsyncInternalKVKeys(
+      const std::string &prefix,
+      const OptionalItemCallback<std::vector<std::string>> &callback) = 0;
+  virtual Status AsyncInternalKVGet(
+      const std::string &key, const OptionalItemCallback<std::string> &callback) = 0;
+  virtual Status AsyncInternalKVPut(const std::string &key, const std::string &value,
+                                    bool overwrite,
+                                    const OptionalItemCallback<int> &callback) = 0;
+  virtual Status AsyncInternalKVExists(const std::string &key,
+                                       const OptionalItemCallback<bool> &callback) = 0;
+  virtual Status AsyncInternalKVDel(const std::string &key,
+                                    const StatusCallback &callback) = 0;
+
+  // These are sync functions of the async above
+  Status Keys(const std::string &prefix, std::vector<std::string> &value);
+  Status Put(const std::string &key, const std::string &value, bool overwrite,
+             bool &added);
+  Status Get(const std::string &key, std::string &value);
```

Comment:
Nit: I don't know if it's best practice, but I'd usually vote for a pointer for this kind of output parameter, e.g.
```suggestion
  Status Get(const std::string &key, std::string *value_out);
```

Reply:
I'll vote 'passing by reference'. In this way, you know the value is not null for sure. If it's input parameter, it'll be const reference, otherwise, just reference. In case you need the 'optional' kind of semantic, I'll pass it by the pointer.

Do you have some comments that why you prefer a pointer?

Reply:
I've generally preferred a pointer since it makes it more obvious at the _call site_ that it's an output parameter:
```
std::string out;
Get("foo", out);
// vs.
Get("foo", &out);
```
However, after a quick read of a few different style guides and Q&As, I agree that since optional semantics aren't needed, passing by reference is the way to go.